es-extension-examples/ext-example-vite/vite.config.js

```diff
@@ -0,0 +1,45 @@
+import reactRefresh from '@vitejs/plugin-react';
+import { defineConfig } from 'vite';
+import dotenv from "dotenv";
+dotenv.config();
+
+const urlPath = process.env.URLPATH || ''
+
+export default defineConfig({
+  /* when in lib mode, must define process.env */
+  define: {
+    'process.env': JSON.stringify({
+      NODE_ENV: process.env.NODE_ENV,
+    })
+  },
+  base: urlPath,
+  root: '.',
+  plugins: [
+    reactRefresh(),
+  ],
+  server: {
+    port: 9098,
+    strictPort: true,
+    host: '0.0.0.0',
+    base: '/',
+  },
+  build: {
+    outDir: 'build',
+    base: urlPath,
+    /* lib config necessary to prevent "Failed to load extension Error" on ext-react-vite */
+    lib: {
```

Comment:
Running Vite in 'lib' mode prevents a "Failed to load extension Error" that occurs for 'ext-react-vite.js'. This is likely happening because in 'lib' mode, Vite configures the output to be more universally compatible with various environments and avoids unnecessary optimizations.